core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.kt

```diff
@@ -147,37 +144,17 @@ open class KiwixWebView @SuppressLint("SetJavaScriptEnabled") constructor(
 
   internal class SaveHandler(private val zimReaderContainer: ZimReaderContainer) :
     Handler() {
-    private fun getDecodedFileName(url: String?, src: String?): String =
-      url?.substringAfterLast("/", "")
-        ?.takeIf { it.contains(".") }
-        ?: src?.substringAfterLast("/", "")
-          ?.substringAfterLast("%3A") ?: ""
 
     @SuppressWarnings("NestedBlockDepth")
     override fun handleMessage(msg: Message) {
       val url = msg.data["url"] as? String
       val src = msg.data["src"] as? String
       if (url != null || src != null) {
-        val fileName = getDecodedFileName(url, src)
-        var root =
-          Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES)
-        if (instance.externalMediaDirs.isNotEmpty()) {
-          root = instance.externalMediaDirs[0]
-        }
-        val fileToSave = sequence {
-          yield(File(root, fileName))
-          yieldAll(generateSequence(1) { it + 1 }.map {
-            File(root, fileName.replace(".", "_$it."))
-          })
-        }.first { !it.exists() }
-        val source = Uri.parse(src)
-        try {
-          zimReaderContainer.load("$source", emptyMap()).data.use { inputStream ->
-            fileToSave.outputStream().use { inputStream.copyTo(it) }
-          }
-          instance.toast(instance.getString(R.string.save_media_saved, fileToSave.name))
-        } catch (e: IOException) {
-          Log.w("kiwix", "Couldn't save image", e)
+        val savedFile = FileUtils.downloadFileFromUrl(url, src, zimReaderContainer)
+        savedFile?.let {
+          instance.toast(instance.getString(R.string.save_media_saved, it.name))
+        } ?: run {
+          Log.w("kiwix", "Couldn't save file")
```

Comment:
We are already doing this in `downoloadFileUrl`, so it's unnecessary here.

Reply:
done